airbyte-integrations/connectors/source-google-ads-asaas/source_google_ads/source.py

```diff
@@ -0,0 +1,231 @@
+#
+# Copyright (c) 2021 Airbyte, Inc., all rights reserved.
+#
+
+
+from logging import exception
+from typing import Any, List, Mapping, Tuple, Union
+
+from airbyte_cdk import AirbyteLogger
+from airbyte_cdk.sources import AbstractSource
+from airbyte_cdk.sources.streams import Stream
+from google.ads.googleads.errors import GoogleAdsException
+from pendulum import timezone
+from pendulum.tz.timezone import Timezone
+import pendulum
+import re
+from dateutil.relativedelta import relativedelta
+
+from .custom_query_stream import CustomQuery
+from .google_ads import GoogleAds
+from .streams import (
+    AccountPerformanceReport,
+    Accounts,
+    AdGroupAdReport,
+    AdGroupAds,
+    AdGroups,
+    Campaigns,
+    CampaignMetricsReport,
+    ClickView,
+    DisplayKeywordPerformanceReport,
+    DisplayTopicsPerformanceReport,
+    GeographicReport,
+    KeywordReport,
+    ShoppingPerformanceReport,
+    UserLocationReport,
+)
+
+
+class PatternHandler(object):
+    """Pattern Handler.
+
+    Class handles the patterns for textual dates
+    """
+
+    def __init__(self, pattern, handler_func) -> None:
+        """Construct Object.
+
+        Args:
+            pattern (str): Regular expression for textual date.
+            handler_func (function): Handler function for transformation.
+        """
+        self.pattern = pattern
+        self.handler_func = handler_func
```

Comment:
```suggestion
        self._pattern = pattern
        self._handler_func = handler_func
```
Utilizar padrão para variáveis privadas. Necessário alterar onde é referenciado.